docs/troubleshooting.md

```diff
@@ -0,0 +1,59 @@
+## Whitelhall troubleshooting
+
+You may run into errors while working locally and trying to render pages on the Frontend.
+
+Problem
+
+```
+Mysql2::Error: Access denied for user 'whitehall'@'localhost' (using password: YES)
+```
+
+Solution:
+Login, create user, give access & quit
+
+```
+mysql -u root -p
+CREATE USER whitehall@localhost IDENTIFIED BY 'whitehall';
+grant all privileges on *.* to whitehall@localhost with grant option;
+\q
+```
+
+Problem
+
+```
+ActiveRecord::SubclassNotFound
+/versions/2.6.6/lib/ruby/gems/2.6.0/gems/sassc-2.4.0/lib/sassc/engine.rb:43: [BUG] Segmentation fault at 0x0000000000000000
+```
+
+Solution: 
+Restart
+
+Problem
+
+```
+/.rvm/rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_gem.rb:67:in `gem'
+/.rvm/rubies/ruby-2.7.2/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_gem.rb:67:in `synchronize': dead
+```
+
+Solution: 
+`gem update --system`
+
+Problem:
+
+```
+API Error
+```
+
+Solution:
+[`search-api`](https://github.com/alphagov/search-api), [`static`](https://github.com/alphagov/static) are helpful to have to running in parallel (depending on what page you are on) via [govuk-docker](https://github.com/alphagov/govuk-docker)
```

Comment:
I know what you're referring to here but could you unpack it for newbies please? Is there a page or a line in the govuk-docker docs which unpack this? All you'd really need is something that says "make sure the referenced apps are up to date as well".

Reply:
Added the key sample commands